### PR TITLE
IGNITE-20178 Introduce param-free IgniteInternalFuture.listen(() -> {}) in addition to .listen((fut) -> {}) to avoid ignored params

### PR DESCRIPTION
**Pull request by anton-vinogradov**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/util/future/GridFinishedFuture.java [hunk 1]
```diff
@@ -138,29 +147,37 @@ public GridFinishedFuture(Throwable err) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public <R> IgniteInternalFuture<R> chain(final IgniteOutClosure<R> doneCb) {
+        return chain(ignored -> doneCb.apply());
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/internal/util/future/GridFinishedFuture.java:152]: `assert doneCb != null;` ?

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java [hunk 2]
```diff
@@ -173,8 +173,8 @@ public class IsolatedDiscoverySpi extends IgniteSpiAdapter implements IgniteDisc
                 msg,
                 null));
 
-            // Acknowledge message must be send after initial message processed.
-            fut.listen((f) -> {
+            // Acknowledge message must be sent after initial message processed.
+            fut.listen(ignored -> {
```
Comment (timoninmaxim) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java:177]: ignored -> () ?

File: modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java [hunk 3]
```diff
@@ -83,7 +81,7 @@ public void testGet() throws Exception {
         try {
             final ComputeTaskFuture<?> fut = executeAsync(ignite.compute(), GridStopTestTask.class.getName(), null);
 
-            fut.listen((CI1<IgniteFuture>)gridFut -> {
+            fut.listen(ignored -> {
```
Comment (timoninmaxim) [inline modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java:84]: ignored -> () ?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java [hunk 4]
```diff
@@ -326,7 +326,7 @@ public void awaitSnpBlockedOrFinished(@Nullable IgniteFuture<?> snpFut) {
             if (latch == null)
                 return;
 
-            snpFut.listen((f) -> latch.countDown());
+            snpFut.listen(ignored -> latch.countDown());
```
Comment (timoninmaxim) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java:329]: ignored -> () ?

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteInternalFuture.java [hunk 5]
```diff
@@ -111,6 +113,14 @@ public interface IgniteInternalFuture<R> {
     @Async.Schedule
     public void listen(IgniteInClosure<? super IgniteInternalFuture<R>> lsnr);
 
+    /**
+     * Registers listener closure to be asynchronously notified whenever future completes.
+     *
+     * @param lsnr Listener closure to register. If not provided - this method is no-op.
```
[the remark this answers is not in the document]
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteInternalFuture.java:119]: We already have such check, see
https://github.com/apache/ignite/blob/4a47dc1ed858b5427e3d27458579242db12639b3/modules/core/src/main/java/org/apache/ignite/internal/util/future/GridFutureAdapter.java#L471

File: modules/core/src/main/java/org/apache/ignite/internal/util/future/GridFinishedFuture.java (hunk 1 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/util/future/GridFinishedFuture.java:152]: Done

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java (hunk 2 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java:177]: It's an IgniteFuture (which is a Public API) that was not changed at this PR.

File: modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java (hunk 3 above)
Comment (anton-vinogradov) [reply modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java:84]: It's an IgniteFuture (which is a Public API) that was not changed at this PR.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java (hunk 4 above)
Comment (anton-vinogradov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java:329]: It's an IgniteFuture (which is a Public API) that was not changed at this PR.

File: modules/core/src/main/java/org/apache/ignite/internal/management/meta/MetadataRemoveTask.java [hunk 6]
```diff
@@ -125,8 +125,8 @@ protected MetadataRemoveJob(@Nullable MetaRemoveCommandArg arg, boolean debug) {
 
                     jobCtx.holdcc();
 
-                    future.listen((IgniteInClosure<IgniteInternalFuture<?>>)f -> {
-                        if (f.isDone())
+                    future.listen(() -> {
+                        if (future.isDone())
```
[the remark this answers is not in the document]
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/management/meta/MetadataRemoveTask.java:129]: Whong comment?

File: modules/core/src/main/java/org/apache/ignite/internal/IgniteInternalFuture.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/IgniteInternalFuture.java]: `If not provided - this method is no-op.` removed

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java (hunk 4 above)
Comment (anton-vinogradov) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/incremental/AbstractIncrementalSnapshotBlockingTest.java]: Rolled back :(

File: modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java (hunk 3 above)
Comment (anton-vinogradov) [reply modules/core/src/test/java/org/apache/ignite/internal/GridTaskFutureImplStopGridSelfTest.java]: Rolled back :(

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java (hunk 2 above)
Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/isolated/IsolatedDiscoverySpi.java]: Rolled back :(

**Review by timoninmaxim — APPROVED** [review]

**Pull request merged**